Styleable: added public member function setRgb()

JaimeIvanCervantes opened the pull request:
Added `setRgb()` function to Styleable.

igagis commented:
Thanks, I understood the intention of the method. Could you fix the review findings please?

JaimeIvanCervantes commented:
Ups, I missed the review. I'm trying to merge master into my branch, but it's blocked because I need the approved review.

igagis commented on src/svgdom/elements/Styleable.cpp:
```diff
@@ -425,6 +425,10 @@ Rgb StyleValue::getRgb() const{
 	return ret;
 }
 
+void StyleValue::setRgb(const std::uint32_t r, const std::uint32_t g, const std::uint32_t b) {
```
yep, this review

igagis commented:
these github reviews are hard to understand, how they work... :).

JaimeIvanCervantes commented:
I updated the code with your changes and I also changed `uint32_t` to `uint8_t` here: https://github.com/igagis/svgdom/blob/master/src/svgdom/elements/Styleable.cpp#L700

JaimeIvanCervantes commented:
hmm tests are failing, there's some weird message about pugixml that seems unrelated to my changes

igagis commented:
It was beacuse cygwin mirror was temporary unavailable, there are such lines in the log:
````
HTTP status 404 fetching http://cygwin.mirror.constant.com/x86_64/setup.xz.sig
````
I restarted the build on Appveyor

igagis commented:
Other builds on travic-ci fail because of some other reason, so there is a problem in the code still.
I also added more review comments, please address those and tests might pass then.

JaimeIvanCervantes commented:
Hmm the tests are failing because `camera.svg` is not matching the etalon sample,but I haven't been able to figure out how this is related to my code. The only changes that I did are removing the `const` and changing `uint32_t` to `uint8_t`.

I just tested my changes on Mac and they work:

```c++
std::unique_ptr<svgdom::SvgElement> dom = utki::makeUnique<svgdom::SvgElement>();

std::unique_ptr<svgdom::PathElement> path = utki::makeUnique<svgdom::PathElement>();

svgdom::PathElement::Step step;

step.type = svgdom::PathElement::Step::Type_e::MOVE_ABS;
step.x = 0;
step.y = 0;
path->path.push_back(step);

step.type = svgdom::PathElement::Step::Type_e::LINE_ABS;
step.x = 0;
step.y = 300;
path->path.push_back(step);

step.type = svgdom::PathElement::Step::Type_e::LINE_ABS;
step.x = 300;
step.y = 300;
path->path.push_back(step);

step.type = svgdom::PathElement::Step::Type_e::LINE_ABS;
step.x = 300;
step.y = 0;
path->path.push_back(step);

StyleValue styleVal;
styleVal.type = StyleValue::Type_e::NORMAL;
//styleVal.color = vcolor[0] | (vcolor[1] << 8) | (vcolor[2] << 16); 
styleVal.setRgb(255, 0, 0); 
path->styles[StyleProperty_e::FILL] = styleVal;

dom->children.push_back(std::move(path));

std::string domStr = dom->toString();
	
std::cout<<domStr<<std::endl;
```
And the outpus is:

```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1">
	<path style="fill:#ff0000" d="M0,0 L0,300 300,300 300,0"/>
</svg>
```

igagis commented:
Instead of `styleVal.setRgb(255, 0, 0); ` try some arbitrary numbers, like `styleVal.setRgb(0xfd, 0x13, 0x4e); ` and see what will be the result of generated `fill:#ff0000`

JaimeIvanCervantes commented:
I changed `uint8_t` back to `uint32_t` and the tests are not failing anymore

igagis commented:
I suggest you to add test for setRgb to make sure it works properly. No need to create a new test, you can add `setRgb` checks to `toString` test

JaimeIvanCervantes commented:
I added the test to `toString` and the tests are passing. I think it's ready to merge.
